lib/app.js

```diff
@@ -404,17 +405,20 @@ module.exports = {
             return value;
           }
         }
-      }];
-
-      getPrompt(argv, schema, function(results) {
-        if (path.relative('.', results.public).match(/^\./)) {
-          console.log(chalk.red('init cancelled - the public directory must be within the current working directory'));
-          process.exit(1);
+      };
+      schemaPublic.conform = function(value) {
+        if (path.relative('.', value).match(/^\./)) {
+          schemaPublic.message = 'Please use a sub directory of current working directory';
```

Comment:
Perhaps something like:

'Please ensure the specified public directory is within the current working directory'